crates/biome_js_analyze/src/assist/source/organize_imports.rs

```diff
@@ -1,143 +1,328 @@
+    /// Only side-effect imports adjacent to a chunk are not separated by a blank line.
+    ///
+    /// Imports or exports of a chunk are sorted together.
+    /// Import and export sorting uses several layer of order.
+    ///
+    /// ## Import and export groups
+    ///
+    /// Imports or exports of a chunk are divided into groups.
+    /// By default, it exists a single group: the implicit group.
+    ///
+    /// You can define extra groups using the `groups` option.
+    /// The organizer provides predefined groups and allows you to create new groups based on a glob or a list of globs.
+    /// The predefined groups are:
     ///
-    /// ```js
-    /// import "../polyfills/array/flatMap";
-    /// import { functionThatUsesFlatMap } from "./utils.js";
+    /// - `:NODE:`: all imports and exports with a source starting by the protocol `node:` or that corresponds to a builtin Node,js module such as `path`
+    /// - `:BUN:`: all imports and exports with a source starting by the protocol `bun:`
+    ///
+    /// Given the following configuration and code, the import organizer propose to reorder the imports to place the Node.js imports first.
+    ///
+    /// ```json,full_options
+    /// {
+    ///     "assist": {
+    ///         "actions": {
+    ///             "source": {
+    ///                 "organizeImports": {
+    ///                     "level": "on",
+    ///                     "options": {
+    ///                         "groups": [
+    ///                             ":NODE:",
+    ///                             ":BUN:"
+    ///                         ]
+    ///                     }
+    ///                 }
+    ///             }
+    ///         }
+    ///     }
+    /// }
     /// ```
     ///
-    /// In these cases, Biome will sort all these three imports, and it might happen that the order will **break** your application.
+    /// ```js,expect_diagnostic,use_options
+    /// import { Database } from "bun:sqlite";
+    /// import lib from "@any/lib";
+    /// import test from "node:test";
+    /// import path from "path";
+    /// ```
     ///
-    /// To avoid this, create a "group" of imports. You create a "group" by adding a **new line** to separate the groups.
+    /// Note that, `import lib from "@any/lib"` doesn't match any explicitly set groups.
+    /// Imports and exports that don't match any user-provided groups, always match the implicit group that appears after all other groups.
+    /// Thus the import is placed at the end.
     ///
-    /// By doing so, Biome will limit the sorting only to the import statements that belong to the same group:
+    /// Groups can also be created using a glob or a list of globs.
+    /// In the following example, we create two groups: one that gathers imports/exports starting with `@any/lib` except `@any/lib/speciaal` and the other that gathers imports/exports starting with `@/`.
```

Comment:
```suggestion
    /// In the following example, we create two groups: one that gathers imports/exports starting with `@any/lib` except `@any/lib/special` and the other that gathers imports/exports starting with `@/`.
```

We should **really explain** what are the semantics of a negated glob here. Like, when the action sees a source that matches `@any/lib/special`, where does it place it? 

Also, another thing that we need to explain is what would happen if we add a glob **after** a negated glob, e.g.: 

```json
{
	"groups": ["@src/**", "!@src/vendor/**", "@src/special/**"]
}
```